src/app/(profile)/(components)/ProfileList/ProfileList.tsx

```diff
@@ -20,13 +22,25 @@ const ProfileList = () => {
   return (
     <div className={styles.profileCards}>
       <div className={styles.profileListCont}>
-        {profiles.map((profile) => (
-          <ProfileListItem
-            key={profile.id}
-            data={profile}
-            href={`${AppRoutes.profile}/${profile.githubUsername}`}
-          />
-        ))}
+        {profiles.map((profile) =>
+          session?.user ? (
+            <ProfileListItem
+              key={profile.id}
+              data={profile}
+              href={`${AppRoutes.profile}/${profile.githubUsername}`}
+            />
+          ) : (
+            <ProfileListItem
+              key={profile.id}
+              data={{
```

Comment:
**nit:** Let's pass props explicitly because not it's not clear what is getting into this component